crates/optimism/payload/src/builder.rs

```diff
@@ -541,3 +556,58 @@ where
 
     Ok(BuildOutcome::Better { payload, cached_reads })
 }
+
+fn get_nonce(
+    is_holocene: bool,
+    attributes: &OptimismPayloadBuilderAttributes,
+    default_base_fee_params: BaseFeeParams,
+) -> B64 {
+    if is_holocene {
+        // If eip 1559 params are set, use them, otherwise use the canyon base fee param constants
+        // The eip 1559 params should exist here since there was a check previously
+        if attributes.eip_1559_params.unwrap() != B64::ZERO {
+            attributes.eip_1559_params.unwrap()
+        } else {
+            let mut default_params = [0u8; 8];
+            default_params[..4].copy_from_slice(
+                &(default_base_fee_params.max_change_denominator as u32).to_be_bytes(),
+            );
+            default_params[4..].copy_from_slice(
+                &(default_base_fee_params.elasticity_multiplier as u32).to_be_bytes(),
+            );
+            B64::from_slice(default_params.as_ref())
+        }
+    } else {
+        BEACON_NONCE.into()
+    }
```

Comment:
This LGTM

Reply:
Same - nice. To clarify, `default_base_fee_params` shouldn't need to be passed. if `eip_1559_params` == 0, we just use canyon base fee params.

Reply:
the `default_base_fee_params` here is just the canyon values from `chain_spec.base_fee_params_at_timestamp(attributes.payload_attributes.timestamp)`

The only way to get the canyon params is via `chain_spec` I think, so need to pass either the params or `chain_spec` to this function

Reply:
related comment above
https://github.com/paradigmxyz/reth/pull/11473/files#r1790622589